src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,102 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
-    @Override
-    public void add(T value) {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elementData;
+    private int size = 0;
 
+    public ArrayList() {
+        elementData = new Object[DEFAULT_CAPACITY];
+    }
+
+    public void add(T value) {
+        grow();
+        elementData[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Cannot add value "
+                    + value + " with index "
+                    + index);
+        }
```

Comment:
Move this to separate method

Reply:
yes, but I have different messages and each one. Is it still worth moving it to a separate method and returning different messages depending on the case?

Reply:
yes, you can give the message as a parameter of exception throwing method